packages/angular_devkit/build_optimizer/src/transforms/scrub-file.ts

```diff
@@ -202,10 +203,22 @@ function isDecorateAssignmentExpression(exprStmt: ts.ExpressionStatement): boole
   }
   const classIdent = expr.left as ts.Identifier;
   const callExpr = expr.right as ts.CallExpression;
+  let callExprIdent = callExpr.expression as ts.Identifier;
+
   if (callExpr.expression.kind !== ts.SyntaxKind.Identifier) {
-    return false;
+    if (callExpr.expression.kind === ts.SyntaxKind.PropertyAccessExpression) {
+      const propAccess = callExpr.expression as ts.PropertyAccessExpression;
+      const left = propAccess.expression;
+      callExprIdent = propAccess.name;
+
+      if (!(left.kind === ts.SyntaxKind.Identifier && (left as ts.Identifier).text === 'tslib')) {
```

Comment:
given `import * as tslib_1 from "tslib";`, we should check if the text matches `tslib(_\d+)?`

or we could trace it back to the import and see if the identifier is an alias for `*` coming from the `"tslib"` module.